docker/puppet-agent-alpine/Dockerfile

```diff
@@ -0,0 +1,137 @@
+FROM alpine:3.8 as build
+
+RUN apk add --no-cache cmake boost-dev make curl git curl-dev ruby ruby-dev gcc g++ yaml-cpp-dev
+
+RUN mkdir /workspace 
+WORKDIR /workspace
+RUN sed -i -e 's/sys\/poll/poll/' /usr/include/boost/asio/detail/socket_types.hpp
+
+RUN git clone -b 1.5.x https://github.com/puppetlabs/leatherman && mkdir -p /workspace/leatherman/build
+WORKDIR /workspace/leatherman/build
+RUN cmake  -DBOOST_STATIC=OFF -DCMAKE_VERBOSE_MAKEFILE=ON ..; make ; make install
```

Comment:
We can defer to a subsequent PR, but shouldn't we be running `; make clean` and doing a few other operations to tidy up the disk contents to remove source / compile time tooling / intermediate output?